src/arrayMethodJoin.js

```diff
@@ -4,8 +4,33 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    // if (typeof separator !== 'string') {
+    //   return this;
+    // }
+    if (this.length === 0) {
+      return '';
+    }
+
+    let result = `${this[0]}`;
```

Comment:
Consider using a ternary operator to initialize 'result' with the appropriate value in one step: let result = (this[0] === null) ? '' : `${this[0]}`;